### PR TITLE
New ctfe get interpret tests to pass

### PR DESCRIPTION
**Pull request by UplinkCoder**

### CODE REVIEW

File: src/bc_common.d [hunk 1]
```diff
@@ -102,12 +102,18 @@ enum BCTypeEnum : ubyte
...
+  Const = 0x1,
+}
 struct BCType
 {
     BCTypeEnum type;
     alias type this;
     /// 0 means basic type
     uint typeIndex;
+    // additional information
+    uint flags;
```
Comment (MartinNowak) [inline src/bc_common.d]: That would later require to transiently apply flags to element types, e.g. to support `const(T*)` and `const(T)*` at the same time.
Not sure why the interpreter has to deal with const though.

Comment (MartinNowak) [inline src/bc_common.d]: A rather annoying test, just to support a bit of casting const away.
Maybe actually modifying the value might be a saner choice than adding a runtime const check.

**Comment by MartinNowak** [comment]
Fairly nice, only a single (but weird) project failure.
`source/vibe/web/internal/rest/common.d(249,10): Error: variable __ctfe forward referenced
source/vibe/web/rest.d(79,14): Error: template instance`

File: src/bc_common.d (hunk 1 above)
Comment (UplinkCoder) [inline src/bc_common.d]: I have no idea, I think the const assignment should be caught before ever getting to the interpreter.

**Pull request merged**